File: tests/python/render_2/render.py
import re
import io
from inc import *

__data = {"title":"Titulo da página",
          "class_1":"container",
          "class_2":"jumbotron",
          "div_1":"conteúdo div 1",
          "div_2":"conteúdo div 2",
          "type_text":"text",
          "type_number":"number",
          "value_input":"valor do primeiro input",
          "value_input_2":"500"}

thisnumber = "variável declarada : thisnumber"
thistext = "variável declarada : thistext"
url = "https://google.com"

file = open("render/render.php", "r")

template = file.read()

pattern = re.compile(r"#{(.*?)}")
pattern_expression = re.compile(r":(.*?);")

selected = pattern.findall(template)
selected_expression = pattern_expression.findall(template)

for j in selected_expression:
    match = pattern_expression.search(template)
    template = template.replace(match.group(0), "<?php echo " + j + "; ?>")

for i in selected:
    match = pattern.search(template)
    template = template.replace(match.group(0), __data[i] if i in __data else eval(i))

#result = pattern.search(template).group(1)
#result = result.replace(" ", "")
#template = pattern.sub(__data[result], template)

file.close()
print(template)

with io.open("render/phase_2.php", "w", encoding = "utf-8") as file:
    file.write(template)

with io.open("public/index.php", "w", encoding = "utf-8") as file:
    file.write(template)

File: tests/python/render_2/render/render.php
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title> #{title} </title>
  </head>
  <body>

      <div class="#{class_1}">

      <div> #{div_1} </div>

      <div class="#{class_2}"> <h2> #{div_2} </h2> </div>

      <input type ="#{type_text}" value = "#{value_input}">

      <input type = "#{type_number}" value = "#{value_input_2}">

      </div>

      <div> #{ "verdadeiro" if 1 < 2 else "falso" } </div>

      <div> #{ thisnumber if len(thisnumber) > len(thistext) else thistext} </div>
      <!-- <div> #{ str(input("insira o conteúdo desta div : "))  </div> -->

      #{ GenerateHtmlElement("div", "este é o conteúdo que vai ser colocado dentro da geração de elemento", "class = 'class_render' ") }

      {{ $data->cep }}
      {{ $data->logradouro }}
      {{ $data->bairro }}


  </body>
</html>
